### PR TITLE
*: remove column mapping from dm docs

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

This PR removes the column mapping content from DM documents because it is not recommended to use column mapping anymore. 

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1899
https://github.com/pingcap/docs/pull/1604

### Which version does your change affect? 

dev, v2.1, v3.0, v3.1

### CODE REVIEW

**Comment by lilin90** [comment]
@TomShawn Please resolve conflicts.

File: dev/how-to/get-started/data-migration.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ $ pgrep -a mysqld
 17782 mysqld --defaults-group-suffix=3
 ```
 
-## Non-overlapping shards
+## Replicate shards
```
Comment (yikeke) [inline dev/how-to/get-started/data-migration.md]: It's preferable to use a heading pattern such as "verb+noun", so this header is correct. But another header "## Starting DM master and workers" in this document does not use the same pattern. We should update it to unify the usage.

Comment (TomShawn) [reply dev/how-to/get-started/data-migration.md]: OK

File: dev/reference/tools/data-migration/usage-scenarios/shard-merge.md [hunk 2]
```diff
@@ -124,31 +124,12 @@ Assume that the downstream schema after replication is as follows:
           tbl-name: "log_bak"
     ```
 
-- To satisfy the replication Requirement #8, configure the [column mapping rule](/dev/reference/tools/data-migration/features/overview.md#column-mapping) as follows:
+- To satisfy the replication Requirement #8, first refer to [handling conflicts of auto-increment primary key](https://github.com/pingcap/docs/pull/1604) to solve conflicts, which guarantees that the data replication does not go wrong when data is replicated to the downstream and the shared table has the same primary key. Then, configure `ignore-checking-items` to skip checking the conflict of auto-increment primary key:
```
Comment (yikeke) [inline dev/reference/tools/data-migration/usage-scenarios/shard-merge.md]:
```suggestion
- To satisfy the replication Requirement #8, first refer to [handling conflicts of auto-increment primary key](https://github.com/pingcap/docs/pull/1604) to solve conflicts. This guarantees that data is successfully replicated to the downstream when the sharded table has duplicate values in the primary key column. Then, configure `ignore-checking-items` to skip checking the conflict of auto-increment primary key:
```
Several things to do:
1. Update the link.
2. In the second sentence, the Chinese "分表中有相同的主键值" means "分表中的主键列有重复值"? Please confirm with the author about this.

**Comment by TomShawn** [comment]
@yikeke PTAL again, thanks.

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**